Utility/UtilityFunctions.hpp

```diff
@@ -0,0 +1,40 @@
+/******************************************************************************
+/ UtilityFunctions.hpp
+/
+/ Copyright (c) 2019
+/
+/ Permission is hereby granted, free of charge, to any person obtaining a copy
+/ of this software and associated documentation files (the "Software"), to deal
+/ in the Software without restriction, including without limitation the rights to
+/ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+/ of the Software, and to permit persons to whom the Software is furnished to
+/ do so, subject to the following conditions:
+/
+/ The above copyright notice and this permission notice shall be included in all
+/ copies or substantial portions of the Software.
+/
+/ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+/ EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+/ OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+/ NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+/ HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+/ WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+/ FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+/ OTHER DEALINGS IN THE SOFTWARE.
+/
+******************************************************************************/
+
+#pragma once
+
+std::string _GetEnvelopeStateChunkBig(TrackEnvelope* envelope, bool isUndo = false);
+
+class _bad_get_env_chunk_big : public std::exception {
+private:
+	std::string msg;
+public:
+	explicit _bad_get_env_chunk_big(const std::string& message) {};
```

Comment:
The message is lost here, what() will always return an empty string. You could make use of std::runtime_error's constructor instead of re-writing it:

```cpp
class bad_get_env_chunk_big : public std::runtime_error {
public:
  using runtime_error::runtime_error;
};
```